flutter/lib/src/user_interaction/sentry_user_interaction_widget.dart

```diff
@@ -0,0 +1,335 @@
+  int? _lastPointerId;
+  Offset? _lastPointerDownLocation;
+  UserInteractionWidget? _lastTappedWidget;
+  ISentrySpan? _activeTransaction;
+
+  Hub get _hub => widget._hub;
+
+  SentryFlutterOptions? get _options =>
+      // ignore: invalid_use_of_internal_member
+      _hub.options as SentryFlutterOptions?;
+
+  @override
+  Widget build(BuildContext context) {
+    return Listener(
+      behavior: HitTestBehavior.translucent,
+      onPointerDown: _onPointerDown,
+      onPointerUp: _onPointerUp,
+      child: widget.child,
+    );
+  }
+
+  void _onPointerDown(PointerDownEvent event) {
+    _lastPointerId = event.pointer;
+    _lastPointerDownLocation = event.localPosition;
+  }
+
+  void _onPointerUp(PointerUpEvent event) {
+    // Figure out if something was tapped
+    final location = _lastPointerDownLocation;
+    if (location == null || event.pointer != _lastPointerId) {
+      return;
+    }
+    final delta = Offset(
+      location.dx - event.localPosition.dx,
+      location.dy - event.localPosition.dy,
+    );
+
+    if (delta.distanceSquared < _tapDeltaArea) {
+      // Widget was tapped
+      _onTappedAt(event.localPosition);
+    }
+  }
+
+  void _onTappedAt(Offset position) {
+    final tappedWidget = _getElementAt(position);
+    final keyValue = tappedWidget?.keyValue;
+    if (tappedWidget == null || keyValue == null) {
+      return;
+    }
+    final element = tappedWidget.element;
+
+    Map<String, dynamic>? data;
+    // ignore: invalid_use_of_internal_member
+    if ((_options?.sendDefaultPii ?? false) &&
+        tappedWidget.description.isNotEmpty) {
+      data = {};
+      data['label'] = tappedWidget.description;
+    }
+
+    const category = 'click';
+    // ignore: invalid_use_of_internal_member
+    if (_options?.enableUserInteractionBreadcrumbs ?? false) {
+      final crumb = Breadcrumb.userInteraction(
+        subCategory: category,
+        viewId: keyValue,
+        viewClass: tappedWidget.type, // to avoid minification
+        data: data,
+      );
+      _hub.addBreadcrumb(crumb, hint: element.widget);
+    }
+
+    // ignore: invalid_use_of_internal_member
+    if (!(_options?.isTracingEnabled() ?? false) ||
+        !(_options?.enableUserInteractionTracing ?? false)) {
+      return;
+    }
+
+    // getting the name of the screen using ModalRoute.of(context).settings.name
+    // is expensive, so we expect that the keys are unique across the app
+    final transactionContext = SentryTransactionContext(
+      keyValue,
+      'ui.action.$category',
+      transactionNameSource: SentryTransactionNameSource.component,
+    );
+
+    final activeTransaction = _activeTransaction;
+    if (activeTransaction != null) {
+      if (_lastTappedWidget?.element.widget == element.widget &&
+          _lastTappedWidget?.eventType == tappedWidget.eventType &&
+          !activeTransaction.finished) {
+        // ignore: invalid_use_of_internal_member
+        activeTransaction.scheduleFinish();
```

Comment:
m: If you're gonna create a new one you dont need to reset the old one timer. Just call `finish` and let `waitForChildren` take care of it.

Reply:
@brustolin not sure I get it, here the user clicked on the same button twice, before the timer was even finished, so we just reset the timer, Android and iOS do the same, its also part of the spec, can you clarify?

Reply:
https://develop.sentry.dev/sdk/performance/ui-event-transactions/

> Scenario: Same UI element with same event
>     Given an ongoing UI event transaction
>     When the user triggers the same UI event with the same type for the same
>         UI element
>     Then the SDK reschedule the idle timeout
>     And doesn't create a new transaction

Reply:
My bad, is the same widget. Gotcha!